include/exadg/solvers_and_preconditioners/preconditioners/preconditioner_amg.h

```diff
@@ -312,17 +312,24 @@ class PreconditionerAMG : public PreconditionerBase<Number>
   void
   vmult(VectorType & dst, VectorType const & src) const final
   {
-    // create temporal vectors of type double
-    VectorTypeAMG dst_amg;
-    dst_amg.reinit(dst, false);
-    VectorTypeAMG src_amg;
-    src_amg.reinit(src, true);
-    src_amg = src;
+    if constexpr(std::is_same_v<Number, NumberAMG>)
+    {
+      preconditioner_amg->vmult(dst, src);
+    }
+    else
+    {
+      // create temporal vectors of type double
```

Comment:
Could you replace all occurrences of `double` in the comments (here, and below) and use `NumberAMG` instead?

Reply:
Done!